game/world/worldobjects.cpp

```diff
@@ -452,10 +452,29 @@ Npc *WorldObjects::findHero() {
   return nullptr;
   }
 
-Npc *WorldObjects::findNpcByInstance(size_t instance) {
-  for(auto& i:npcArr)
-    if(i->handle().symbol_index()==instance)
-      return i.get();
+Npc *WorldObjects::findNpcByInstance(size_t instance, uint32_t n) {
+  if(n==0)
```

Comment:
Normally in C++ indexing is normally zero-based. 
I'm suggesting to have this function as zero-based, and in that case `-1` passed from caller will be casted to `UINT_MAX` and result into null result.